### PR TITLE
fix(44448): Non-optional but undefined delete under exactOptionalPropertyTypes

### LINKED ISSUE #44448: Non-optional but undefined `delete` under strictOptionalProperties
### 🔎 Search Terms

Optional operand delete strictOptionalProperties

### 🕗 Version & Regression Information

- I was unable to test this on prior versions because **the flag is new in 4.4**

### ⏯ Playground Link

[Playground link with relevant code](https://www.typescriptlang.org/play?strictOptionalProperties=true&ts=4.4.0-dev.20210602#code/CYUwxgNghgTiAEYD2A7AzgF3kgRgKwC54BvUAMwEsUKMQIBPACRDgFkp6cQBVFcqkMCIoArgFsuMeAB94IviEopBAXwDcAKA2gIIWtnwA6ftVoNmbDl14nBmoA)

### 💻 Code

```ts
declare const obj: {definitelyHereMaybeUndefined: number | undefined};

delete obj.definitelyHereMaybeUndefined;
```

### 🙁 Actual behavior

No type check diagnostic was produced.

### 🙂 Expected behavior

Since strictOptionalProperties distinguishes between something being able to be `undefined` and it being optional, the type checker should produce a `The operand of a 'delete' operator must be optional. (2790)` diagnostic.

### PR DESCRIPTION
**Pull request by a-tarasyuk**
Fixes #44448

### CODE REVIEW

**Comment by DanielRosenwasser** [comment]
You may need to rewrite a few tests since this option is going to be renamed and removed from `--strict`.

**Comment by a-tarasyuk** [comment]
@DanielRosenwasser Do you mean to use `strictOptionalProperties` in order to fix it after rename?

**Comment by DanielRosenwasser** [comment]
Yeah, and tests that assume `strict` will toggle the option will change in their behavior.

**Review by DanielRosenwasser — APPROVED** [review]

File: tests/baselines/reference/deleteExpressionMustBeOptional_exactOptionalPropertyTypes(exactoptionalpropertytypes=true).errors.txt [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+!!! error TS2790: The operand of a 'delete' operator must be optional.
+    delete f.b
+           ~~~
+!!! error TS2790: The operand of a 'delete' operator must be optional.
+    delete f.c
+           ~~~
+!!! error TS2790: The operand of a 'delete' operator must be optional.
+    delete f.d
+           ~~~
+!!! error TS2790: The operand of a 'delete' operator must be optional.
```
Comment (DanielRosenwasser) [inline tests/baselines/reference/deleteExpressionMustBeOptional_exactOptionalPropertyTypes(exactoptionalpropertytypes=true).errors.txt:46]: This was a bug - we need to check whether the property is optional.

https://github.com/microsoft/TypeScript/issues/44852

Comment (a-tarasyuk) [reply tests/baselines/reference/deleteExpressionMustBeOptional_exactOptionalPropertyTypes(exactoptionalpropertytypes=true).errors.txt:46]: Should `f.f` be ignored too? Or am I wrong?

**Pull request merged**